security/security-spring/src/main/java/org/cbioportal/security/spring/authentication/keycloak/SAMLUserDetailsServiceImpl.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package org.cbioportal.security.spring.authentication.keycloak;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.cbioportal.security.spring.authentication.PortalUserDetails;
+
+import org.opensaml.saml2.core.Attribute;
+
+import org.opensaml.xml.XMLObject;
+import org.opensaml.xml.schema.XSString;
+import org.opensaml.xml.schema.impl.XSAnyImpl;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.authority.AuthorityUtils;
+import org.springframework.security.saml.SAMLCredential;
+import org.springframework.security.saml.userdetails.SAMLUserDetailsService;
+import org.springframework.stereotype.Service;
+
+/**
+ * Implements spring security SAMLUserDetailsService interface:
+ * Instantiates PortalUserDetails object from SAML assertion received from
+ * keycloak identity provide. Creates Spring Security GrantedAuthority list 
+ * based on cancer study roles included in the assertion during authentication
+ *
+ * @author Kelsey Zhu
+ */
+
+@Service
+public class SAMLUserDetailsServiceImpl implements SAMLUserDetailsService
+{
+    private static final Log log = LogFactory.getLog(SAMLUserDetailsServiceImpl.class);
+
+    @Value("${saml.idp.metadata.attribute.email:}")
+    private String SAML_IDP_METADATA_EMAIL_ATTR_NAME;
+
+    @Value("${saml.idp.metadata.attribute.role:}")
+    private String SAML_IDP_METADATA_ROLE_ATTR_NAME;
+
+    @Value("${app.name:}")
+    private String APP_NAME;
+    
+
+    /**
+     * Default no_arg Constructor.
+     */
+    public SAMLUserDetailsServiceImpl() {
+    }
+
+    /**
+     * 
+     * @param attributeValue
+     * @return String type attribute value
+     */
+    private String getAttributeValue(XMLObject attributeValue)
+    {
+        return attributeValue == null ?
```

Comment:
It's hard to debug statements like this. Can we break it out into a few if-else blocks?